### ISSUE TITLE
TypeError: _validate_attrs() missing 1 required positional argument: 'engine'

### ISSUE THREAD

**Issue by jhamman** [ecmwf/cfgrib#400]
### What happened?

An upstream change to xarray's `validate_attrs` function has broken `canonical_dataset_to_grib`. This change has not been released yet but will in the coming days. 

See https://github.com/pydata/xarray/pull/9407 for more details.

### What are the steps to reproduce the bug?

The traceback is below:

```
    def canonical_dataset_to_grib(dataset, path, mode="wb", no_warn=False, grib_keys={}, **kwargs):
        # type: (xr.Dataset, str, str, bool, T.Dict[str, T.Any], T.Any) -> None
        """
        Write a ``xr.Dataset`` in *canonical* form to a GRIB file.
        """
        if not no_warn:
            warnings.warn("GRIB write support is experimental, DO NOT RELY ON IT!", FutureWarning)
    
        # validate Dataset keys, DataArray names, and attr keys/values
        xr.backends.api._validate_dataset_names(dataset)  # type: ignore
>       xr.backends.api._validate_attrs(dataset)  # type: ignore
E       TypeError: _validate_attrs() missing 1 required positional argument: 'engine'
```

### Version

0.9.14.0

### Platform (OS and architecture)

linux

**Comment by iainrussell** [comment]
Thank you @jhamman - that was a useful link, and sped up the fix! I'm just releasing 0.9.14.1 now with this fix.

**Comment by jhamman** [comment]
Thanks @iainrussell!
